pkgdb/src/search/command.cc

```diff
@@ -181,13 +181,52 @@ SearchCommand::run()
       std::cout << query.str() << std::endl;
       return EXIT_SUCCESS;
     }
+  auto                                            resultCount = 0;
+  std::vector<std::vector<pkgdb::row_id>>         ids;
+  std::vector<std::shared_ptr<pkgdb::PkgDbInput>> inputs;
   for ( const auto & [name, input] :
         *this->getEnvironment().getPkgDbRegistry() )
     {
-      auto dbRO = input->getDbReadOnly();
-      for ( const auto & row : query.execute( dbRO->db ) )
+      auto                       dbRO = input->getDbReadOnly();
+      std::vector<pkgdb::row_id> inputIds;
+      for ( const auto & id : query.execute( dbRO->db ) )
         {
-          std::cout << input->getRowJSON( row ).dump() << std::endl;
+          inputIds.emplace_back( id );
+          resultCount += 1;
+        }
+      inputs.emplace_back( input );
+      ids.emplace_back( std::move( inputIds ) );
+    }
+  if ( query.limit.has_value() )
```

Comment:
nonblocking: it seems like it would be more correct and performant to add a `LIMIT` and a `COUNT` to the actual SQL query executed rather than implementing that functionality in C++

Reply:
nonblocking just discussing:
Yeah I hadn't understood the multiple db stuff. 
- I don't think `limit` should be a field of `PkgQuery` if it's not used to limit results. It should only be a field of `SearchParams` or `SearchQuery`.
- It would still be more performant to do a `LIMIT` and a `COUNT` on the query against each database (which I believe you can do in a single query), and then you can sum the counts. `databases * limit` is potentially a lot less than `databases * entries in each database`

Reply:
> I don't think limit should be a field of PkgQuery if it's not used to limit results. It should only be a field of SearchParams or SearchQuery.

That's fair, but I'm not sure it makes much of a difference at this stage.

> `databases * limit` is potentially a lot less than `databases * entries` in each database

Also fair, but the database queries aren't really a performance bottleneck at the moment. There's some post-GA search work planned (namely semver filtering in SQL) that will focus on query performance and probably end up rewriting these queries entirely.